src/stdlib_io_filesystem.F90

```diff
@@ -0,0 +1,181 @@
+! SPDX-Identifier: MIT
+
+ !> Interaction with the filesystem.
+module stdlib_io_filesystem
+    use stdlib_string_type, only: string_type
+    implicit none
+    private
+
+    public :: temp_dir, is_windows, exists, path_separator, list_dir, mkdir, rmdir, run
+
+    character(*), parameter :: temp_dir = 'temp'
+
+contains
+
+    !> Version: experimental
+    !>
+    !> Whether the operating system is Windows.
+    !> [Specification](../page/specs/stdlib_io.html#is_windows)
+    logical function is_windows()
```

Comment:
In [fpm](https://github.com/fortran-lang/fpm/blob/da6d1bf89a241cd87d640b889ab1e570e8be5f35/src/fpm_environment.f90#L66), the method of using local `save` variables seems very advisable to avoid repeatedly accessing environment variables when `is_windows` is frequently called.